e2store/src/era.rs

```diff
@@ -44,15 +44,25 @@ impl Era {
         let version = VersionEntry::try_from(&file.entries[0])?;
         let entries_length = file.entries.len();
         let mut blocks = vec![];
+
+        let slot_index_block = SlotIndexBlockEntry::try_from(&file.entries[entries_length - 2])?;
+        let slot_index_state = SlotIndexStateEntry::try_from(&file.entries[entries_length - 1])?;
+
         // Iterate over the block entries. Skip the first and last 3 entries.
+        let mut next_slot = slot_index_block.slot_index.starting_slot;
         for idx in 1..entries_length - 4 {
             let entry: Entry = file.entries[idx].clone();
-            let beacon_block = CompressedSignedBeaconBlock::try_from(&entry)?;
+            let fork = get_beacon_fork(next_slot);
+            let beacon_block = CompressedSignedBeaconBlock::try_from(&entry, fork)?;
+            next_slot = match &beacon_block.block {
+                SignedBeaconBlock::Bellatrix(block) => block.message.slot,
+                SignedBeaconBlock::Capella(block) => block.message.slot,
+                SignedBeaconBlock::Deneb(block) => block.message.slot,
+            } + 1;
```

Comment:
nit:

```suggestion
            next_slot = match &beacon_block.block {
                SignedBeaconBlock::Bellatrix(block) 
                | SignedBeaconBlock::Capella(block) 
                | SignedBeaconBlock::Deneb(block) => block.message.slot,
            } + 1;
```

Reply:
I tried using this, but it gave me an error because the enum returns different types.

I will make a getter in `SignedBeaconBlock` should should simply this common case